h2o-automl/src/main/java/ai/h2o/automl/leaderboard/Leaderboard.java

```diff
@@ -95,58 +160,8 @@ public Leaderboard(String project_name, EventLog eventLog, Frame leaderboardFram
     _sort_metric = sort_metric == null ? null : sort_metric.toLowerCase();
   }
 
-  /**
-   * Retrieves a leaderboard from DKV or creates a fresh one and add it to DKV.
-   *
-   * Note that if the leaderboard is reused to add new models, we have to use the same leaderboard frame.
-   *
-   * IMPORTANT!
-   * if the leaderboard is created without leaderboardFrame, the models will be sorted according to their default metrics
-   * (in order of availability: cross-validation metrics, validation metrics, training metrics).
-   * Therefore, if some models were trained with/without cross-validation, or with different training or validation frames,
-   * then we can't guarantee the fairness of the leaderboard ranking.
-   *
-   * @param project_name
-   * @param eventLog
-   * @param leaderboardFrame
-   * @param sort_metric
-   * @return an existing leaderboard if there's already one in DKV for this project, or a new leaderboard added to DKV.
-   */
-  static Leaderboard getOrMake(String project_name, EventLog eventLog, Frame leaderboardFrame, String sort_metric) {
-    Leaderboard leaderboard = DKV.getGet(Key.make(idForProject(project_name)));
-    if (null != leaderboard) {
-      if (leaderboardFrame != null
-              && (!leaderboardFrame._key.equals(leaderboard._leaderboard_frame_key)
-                  || leaderboardFrame.checksum() != leaderboard._leaderboard_frame_checksum)) {
-        throw new H2OIllegalArgumentException("Cannot use leaderboard "+project_name+" with a new leaderboard frame"
-                +" (existing leaderboard frame: "+leaderboard._leaderboard_frame_key+").");
-      } else {
-        eventLog.warn(Stage.Workflow, "New models will be added to existing leaderboard "+project_name
-                +" (leaderboard frame="+leaderboard._leaderboard_frame_key+") with already "+leaderboard.getModelKeys().length+" models.");
-      }
-      if (sort_metric != null && !sort_metric.equals(leaderboard._sort_metric)) {
-        leaderboard._sort_metric = sort_metric.toLowerCase();
-        if (leaderboard.getLeader() != null) leaderboard.setDefaultMetrics(leaderboard.getLeader()); //reinitialize
-      }
-    } else {
-      leaderboard = new Leaderboard(project_name, eventLog, leaderboardFrame, sort_metric);
-    }
-    DKV.put(leaderboard);
-    return leaderboard;
-  }
-
-  /**
-   * @param project_name
-   * @return a Leaderboard id for the project name
-   */
-  public static String idForProject(String project_name) { return "AutoML_Leaderboard_" + project_name; }
-
-  /**
-   * @param metric
-   * @return true iff the metric is a loss function
-   */
-  public static boolean isLossFunction(String metric) {
-    return metric != null && !Arrays.asList("auc", "aucpr").contains(metric.toLowerCase());
+  public void setExtensionsProvider(LeaderboardExtensionsProvider provider) {
```

Comment:
@seb-h2o except for the docs what would suggest user to use this method? maybe add example to javadoc of Leaderboard as you did for Py/R?

Reply:
adding more javadoc in various places now.